### PR TITLE
Replace `CertificatesSupportedInEra` with `ShelleyBasedEra`

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Delete `CertificatesSupportedInEra`.  Use `ShelleyBasedEra` instead.
    Delete `certificatesSupportedInEra`.  Use `inEonForEra` instead.
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-api/internal/Cardano/Api/TxBody.hs [hunk 1]
```diff
@@ -3210,59 +3180,11 @@ fromLedgerTxCertificates
...
-      where
-        certificates = body ^. L.certsTxBodyL
-
-    -- TODO: Implement once certificates are done in Conway.
-    ShelleyBasedEraConway -> TxCertificatesNone
+  shelleyBasedEraConstraints sbe $
+    let certificates = body ^. L.certsTxBodyL in
+    if null certificates
+      then TxCertificatesNone
+      else TxCertificates sbe (map (fromShelleyCertificate sbe) $ toList certificates) ViewTx
```
Comment (newhoggy) [inline cardano-api/internal/Cardano/Api/TxBody.hs:3187]: This function is simplified via the use of `shelleyBasedEraConstraints`.

File: cardano-api/internal/Cardano/Api/TxBody.hs [hunk 2]
```diff
@@ -1687,13 +1656,14 @@ deriving instance Show (TxWithdrawals build era)
 
 data TxCertificates build era where
```
Comment (carbolymer) [inline cardano-api/internal/Cardano/Api/TxBody.hs:1657]: Can we replace this with 
```suggestion
type TxCertificates build era 
  = Featured ShelleyBasedEra era 
    ( [Certificate era]
    , BuildTxWith build (Map StakeCredential (Witness WitCtxStake era))
    )
```
?

**Review by carbolymer — APPROVED** [review]

File: cardano-api/internal/Cardano/Api/TxBody.hs (hunk 2 above)
Comment (newhoggy) [reply cardano-api/internal/Cardano/Api/TxBody.hs:1657]: Yep.  Can do that in a future commit.

Comment (Jimbo4350) [reply cardano-api/internal/Cardano/Api/TxBody.hs:1657]: Please don't. Having this as a data definition makes it clearer what it is.

**Pull request merged**